Music app/app/src/main/java/com/example/musicapp/NotificationPlaying/MusicService.java

```diff
@@ -0,0 +1,209 @@
+package com.example.musicapp.NotificationPlaying;
+
+
+import android.app.Notification;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.app.Service;
+import android.content.Context;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.media.MediaPlayer;
+import android.net.Uri;
+import android.os.Binder;
+import android.os.IBinder;
+import android.support.v4.media.session.MediaSessionCompat;
+import android.util.Log;
+import android.widget.Toast;
+
+import androidx.annotation.Nullable;
+import androidx.core.app.NotificationCompat;
+
+import com.example.musicapp.Activities.PlayerActivity;
+import com.example.musicapp.Musics.MusicAdapter;
+import com.example.musicapp.Musics.MusicFile;
+import com.example.musicapp.R;
+
+import java.util.ArrayList;
+
+public class MusicService extends Service implements MediaPlayer.OnCompletionListener {
+    IBinder mBinder = new MyBinder();
+    public static MediaPlayer mediaPlayer;
+    ArrayList<MusicFile> musicFiles = new ArrayList<>();
+    Uri uri;
+    int position = -1;
+    ActionPlaying actionPlaying;
+    MediaSessionCompat mediaSessionCompat;
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        mediaSessionCompat = new MediaSessionCompat(getBaseContext(), "My Audio");
+    }
+
+    @Nullable
+    @Override
+    public IBinder onBind(Intent intent) {
+        Log.e("Bind", "Method");
+        return mBinder;
+    }
+
+    public void pause() {
+        mediaPlayer.pause();
+    }
+
+
+    public class MyBinder extends Binder {
+        public MusicService getService() {
+            return MusicService.this;
+        }
+    }
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        int myPosition = intent.getIntExtra("servicePosition", -1);
+        String actionName = intent.getStringExtra("ActionName");
+        int sameSong = intent.getIntExtra("sameSong", 0);
+        if (myPosition != -1 && sameSong == 0)
+            playMedia(myPosition);
+        if (actionName != null) {
+            switch (actionName) {
+                case "playPause":
+                    if (actionPlaying != null) {
+                        Log.e("Inside", "ActionPlayPause");
+                        actionPlaying.playPauseBtnClicked();
+                    }
+                    break;
+                case "next":
+                    if (actionPlaying != null) {
+                        Log.e("Inside", "ActionNext");
+                        actionPlaying.nextBtnClicked();
+                    }
+                    break;
+                case "previous":
+                    if (actionPlaying != null) {
+                        Log.e("Inside", "ActionPrevious");
+                        actionPlaying.prevBtnClicked();
+                    }
+                    break;
+
+            }
+        }
+        return START_STICKY;
```

Comment:
What is START_STICKY?

Reply:
I get it from [stackoverflow](https://stackoverflow.com/questions/9093271/start-sticky-and-start-not-sticky)

Reply:
OK